src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,85 @@
 package core.basesyntax;
 
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.List;
+
 public class WorkWithFile {
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String RESULT = "result";
+    private static final char COMMA = ',';
+    private int countSupply;
+    private int countBuy;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        setCountSupply(0);
+        setCountBuy(0);
+
+        File file = new File(fromFileName);
+        try {
```

Comment:
https://github.com/mate-academy/jv-work-with-file/blob/master/checklist.md#divide-program-logic

Please split logic of public method.
You may have 3 methods: 1st to read file conntent, 2nd to form report String and 3rd to write it to file